backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/geocoding/query.kt

```diff
@@ -0,0 +1,72 @@
+package app.ehrenamtskarte.backend.stores.geocoding
+
+import app.ehrenamtskarte.backend.common.COUNTRY_CODE
+import app.ehrenamtskarte.backend.common.STATE
+import app.ehrenamtskarte.backend.stores.importer.types.AcceptingStore
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.ktor.client.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import org.geojson.Feature
+import org.geojson.FeatureCollection
+import java.io.File
+
+suspend fun queryFeatures(params: List<Pair<String, String>>): List<Feature> {
+    val fileName = cacheFileName(params)
+    val file = File("${System.getProperty("user.dir")}/data/nominatim/v1/$fileName.json")
+
+    val geoJson = if (file.exists()) {
+        // Use a cached version on disk.
+        file.readText()
+    } else {
```

Comment:
Instead of implementing a cache manually, which can be hard and error prone I would suggest to use the cache provided by the http library: https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client.plugins.cache/-http-cache/index.html?query=class%20HttpCache

This cache respects the HTTP caching headers. This means we can configure from the nominatim/geocoder side how long responses should stay valid. Everything is handled transparently by the http client library.

This approach also does not mix concerns like here. At least a cache should be extracted from this function.

Reply:
Do you have some more resources on how to use the ktor cache?

Reply:
I guess the cache i linked is not sufficient. Maybe there other options though.

Reply:
Possibly it is easy to implement the following interface: https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client.plugins.cache.storage/-http-cache-storage/index.html

Reply:
#445